### ISSUE TITLE
emqx Dual-Node Cluster Split-Brain Issue due to Network

### ISSUE THREAD

**Issue by szclouds** [emqx/emqx#11157]
### What happened?

**Overview**
Cluster Mode: Two machines, namely [emqx@172.18.0.8](mailto:emqx@172.16.0.8) and [emqx@172.18.0.](mailto:emqx@172.16.0.8)9, with each node located on different servers, connected via a switch.

emqx Version: EMQ X Broker 4.3.9

**Issue**
When the network of the [emqx@172.18.0.](mailto:emqx@172.16.0.8)9 node encounters a failure (the network cable of that node is manually disconnected), lasting approximately 10 minutes, and then the cable is plugged back in, the following situation persists after network recovery:

From the perspective of the emqx@172.18.0.8 node:

```
Cluster status: #{running_nodes => ['emqx@172.18.0.8'],
                  stopped_nodes => []}
```
From the perspective of the emqx@172.18.0.9 node:

```
Cluster status: #{running_nodes => ['emqx@172.18.0.9'],
                  stopped_nodes => []}
```
The expected normal situation should be:

```
Cluster status: #{running_nodes => ['emqx@172.18.0.8','emqx@172.18.0.9'],
                  stopped_nodes => []}
```
**The node logs before and after network recovery are as follows:**

emqx@172.18.0.8 node:

```
2023-06-28T09:28:49.011380+00:00 [error] ** Node 'emqx@172.18.0.9' not responding **, ** Removing (timedout) connection **
2023-06-28T09:38:19.070057+00:00 [warning] [Alarm Handler] Alarm high_cpu_usage is activated, 82.01209271078267% cpu usage
2023-06-28T09:39:19.072258+00:00 [warning] [Alarm Handler] Alarm high_cpu_usage is deactivated
2023-06-28T09:48:15.434050+00:00 [critical] Ekka(Monitor): Network partition detected from node emqx@172.18.0.9: running_partitioned_network
2023-06-28T09:48:15.434029+00:00 [error] Mnesia('emqx@172.18.0.8'): ** ERROR ** mnesia_event got {inconsistent_database, running_partitioned_network, 'emqx@172.18.0.9'}
2023-06-28T09:48:15.434345+00:00 [warning] [Alarm Handler] Alarm partition is activated, Partition occurs at node emqx@172.18.0.9
2023-06-28T09:50:17.231265+00:00 [error] Mnesia('emqx@172.18.0.8'): ** ERROR ** mnesia_event got {inconsistent_database, running_partitioned_network, 'emqx@172.18.0.9'}
2023-06-28T09:50:17.231375+00:00 [critical] Ekka(Monitor): Network partition detected from node emqx@172.18.0.9: running_partitioned_network
2023-06-28T09:50:17.323672+00:00 [warning] [Alarm Handler] Alarm partition is deactivated
2023-06-28T10:57:29.364194+00:00 [error] 172.18.0.9:56864 [MQTT] , Parse failed for function_clause,......
```
emqx@172.18.0.9 node:
```
2023-06-28T09:28:53.148862+00:00 [error] ** Node 'emqx@172.18.0.8' not responding **, ** Removing (timedout) connection **
2023-06-28T09:42:53.331264+00:00 [warning] [Alarm Handler] Alarm high_cpu_usage is deactivated
2023-06-28T09:48:15.483117+00:00 [error] Mnesia('emqx@172.18.0.9'): ** ERROR ** mnesia_event got {inconsistent_database, running_partitioned_network, 'emqx@172.18.0.8'}
2023-06-28T09:48:15.483262+00:00 [critical] Ekka(Monitor): Network partition detected from node emqx@172.18.0.8: running_partitioned_network
2023-06-28T09:48:15.483421+00:00 [warning] [Alarm Handler] Alarm partition is activated, Partition occurs at node emqx@172.18.0.8
```
**Temporary Solution**
After restarting the emqx@172.18.0.9 node, the issue is resolved.

**In summary, is this a configuration issue with my emqx setup? Or do I need to upgrade to a higher version?**

### What did you expect to happen?

emqx@172.18.0.8 node and emqx@172.18.0.9 node:
```
Cluster status: #{running_nodes => ['emqx@172.18.0.8','emqx@172.18.0.9'],
                  stopped_nodes => []}
```

### EMQX version

EMQ X Broker 4.3.9

### OS version

Ubuntu
18.04.5 LTS (Bionic Beaver)

**Comment by savonarola** [comment]
Hi!
v4.3.9 is quite an old version, and the whole 4.3 version is unsupported.
Could you try to upgrade to the latest 4.4 version?

**Comment by szclouds** [comment]
> Hi! v4.3.9 is quite an old version, and the whole 4.3 version is unsupported. Could you try to upgrade to the latest 4.4 version?
@savonarola 
Okay, thank you for your reply. I will upgrade to v4.4.18 and give it a try.
